### DISCUSSION TITLE
Realm JavaScript v10.20.0-beta.1

### DISCUSSION THREAD

**Discussion by kraenhansen** [realm/realm-js#4275] [General]
### Breaking change
* Removed all code related to the legacy Chrome Debugger. Please use [Flipper](https://fbflipper.com/) as debugger.

### Enhancements
* `Realm.writeCopyTo()` now supports creating snapshots of synced Realms, thus allowing apps to be shipped with partially-populated synced databases. ([#3782](https://github.com/realm/realm-js/issues/3782)
* Added beta support for flexible sync ([#4220](https://github.com/realm/realm-js/pull/4220)).
* Adding support for passing the `arrayFilters` parameter when updating documents using the remote MongoDB client. ([#4248](https://github.com/realm/realm-js/pull/4248))

### Fixed
* Fixed "JSCRuntime destroyed with a dangling API object" assertion when reloading an app in debug mode while running with Hermes engine disabled. ([#4115](https://github.com/realm/realm-js/issues/4115), since 10.20.0-alpha.0)

### Compatibility
* MongoDB Realm Cloud.
* Realm Studio v11.0.0.
* APIs are backwards compatible with all previous releases of Realm JavaScript in the 10.5.x series.
* File format: generates Realms with format v22 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).

### Internal
* Remove the previous implementation to the JavaScriptCore engine (in `src/jsc`).

<hr /><em>This discussion was created from the release <a href='https://github.com/realm/realm-js/releases/tag/v10.20.0-beta.1'>Realm JavaScript v10.20.0-beta.1</a>.</em>

**Comment by ykravv** [comment]
Hey guys, thanks for the release :clap:

I got lost between the issue and Hermes closed PR. Do you have insight into when it's planned to release a stable version with Hermes support or is there a roadmap for this?

**Comment by aleminkirali** [comment]
@kraenhansen Any update when we can expect stable release with Hermes? It's weird that you have v10.20.0-beta.1 then skip this and bring out stable version of v10.13.

**Reply by leecommamichael** [reply]
How is that weird?

**Reply by ianpward** [reply]
Hey @aleminkirali - Product for Realm here, while there are not too many user-facing changes in this release, to support Hermes there was a significant amount of code refactored under the hood to support this and so we plan to run a long-running beta to gather user feedback and confidence with the implementation. This will likely last several months. Of course, it's really your call if you'd like to use a beta library in your production app - and already some users do - but only you, as the app developer, can make the call on whether the performance and stability work for your app.

**Reply by leecommamichael** [reply]
>  This will likely last several months.

Ouch. Thanks for clarifying that timeline. This is such an awkward time as everyone migrates away from JSC.

**Comment by Acetyld** [comment]
Any idea about this release?  React native is pushing hard to merging to hermes default https://reactnative.dev/blog/2022/03/15/an-update-on-the-new-architecture-rollout

**Comment by HSReact** [comment]
Not using Hermes in RN forces us to make too many compromises e.g. not being able to use reanimated 2 etc.

I hope the Realm team can deliver a stable release soon.
